packages/react-server/core/__tests__/ClientRequestSpec.js

```diff
@@ -7,6 +7,24 @@ describe("ClientRequest", () => {
 		clientRequest = new ClientRequest("/");
 	});
 
+	it("removes fragment identifiers from the url", (done) => {
+		clientRequest = new ClientRequest("/");
+		expect(clientRequest.getUrl()).toEqual("/");
+		clientRequest = new ClientRequest("/react-server/foo#bar");
+		expect(clientRequest.getUrl()).toEqual("/react-server/foo");
+		clientRequest = new ClientRequest("/react-server/foo/#bar#bazz");
+		expect(clientRequest.getUrl()).toEqual("/react-server/foo/");
+		clientRequest = new ClientRequest("/react-server/foo/?#bar");
+		expect(clientRequest.getUrl()).toEqual("/react-server/foo/?");
+		clientRequest = new ClientRequest("/react-server/foo/?foo=bar&baz=123");
+		expect(clientRequest.getUrl()).toEqual("/react-server/foo/?foo=bar&baz=123");
+		clientRequest = new ClientRequest("/react-server/foo/?foo=bar&baz=123#");
+		expect(clientRequest.getUrl()).toEqual("/react-server/foo/?foo=bar&baz=123");
+		clientRequest = new ClientRequest("/react-server/foo/?foo=bar&baz=123&zed=abc?#some-fragment?#");
+		expect(clientRequest.getUrl()).toEqual("/react-server/foo/?foo=bar&baz=123&zed=abc?");
+		done();
+	});
```

Comment:
Thanks for adding tests!

One additional test I'd like to see here is a test that has a URL with a `?` following the `#`. e.g. `/react-server/foo#?bar=3&foo=7`. I feel like I've seen libraries that store data in the hash in that format, and because we're looking for both `#` and `?` in regexes in `ClientRequest`, it seems like a place we could have regressions in the future.

Reply:
In this case, what would the expected behavior be? Based on this:

https://en.wikipedia.org/wiki/Uniform_Resource_Identifier#Syntax

It seems like it will be wrong to store the part after the `?` as as query params in this case since it does not comply with the uri syntax.

Reply:
Right, I'd expect that everything after the `#` is stripped -- it's still part of the hash, and not the query string. The concern would be that something in the code changes later, and all of a sudden we start matching the `?` before the `#` or something, and it _accidentally_ gets included as the query string. A test case should help prevent that kind of regression.